### PR TITLE
[Lambda] Add operation to list functions, list function version and to delete functions

### PR DESCRIPTION
**Pull request by Nyholm**
Im creating a small app to delete layer functions not used.

### CODE REVIEW

File: src/Service/Lambda/src/Enum/Runtime.php [hunk 1]
```diff
@@ -3,7 +3,7 @@
 namespace AsyncAws\Lambda\Enum;
 
 /**
- * A runtime identifier. For example, `go1.x`.
+ * The runtime environment for the Lambda function.
```
Comment (jderusse) [inline src/Service/Lambda/src/Enum/Runtime.php:6]: why isn't it updated by the bot?

Comment (Nyholm) [reply src/Service/Lambda/src/Enum/Runtime.php:6]: Maybe it is. Let's see tomorrow when the bot runs next time.

Comment (Nyholm) [reply src/Service/Lambda/src/Enum/Runtime.php:6]: hm.. No update by the bot..

Strange.

**Comment by Nyholm** [comment]
The PR is updated and ready for review. 

I generated 3 operations and added tests. I also added a changelog as I prepare a release after merging this.

File: src/Service/Lambda/src/Enum/Runtime.php (hunk 1 above)
Comment (jderusse) [reply src/Service/Lambda/src/Enum/Runtime.php:6]: Ok, I know why.
We don't have documentation for Shapes. We only have documentation when a Shape is referenced by an attribute.

For the Runtime Shape, we have differents documentations:
```
"Runtime": {
      "base": null,
      "refs": {
        "CompatibleRuntimes$member": null,
        "CreateFunctionRequest$Runtime": "<p>The identifier of the function's <a href=\"https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html\">runtime</a>.</p>",
        "FunctionConfiguration$Runtime": "<p>The runtime environment for the Lambda function.</p>",
        "ListLayerVersionsRequest$CompatibleRuntime": "<p>A runtime identifier. For example, <code>go1.x</code>.</p>",
        "ListLayersRequest$CompatibleRuntime": "<p>A runtime identifier. For example, <code>go1.x</code>.</p>",
        "UpdateFunctionConfigurationRequest$Runtime": "<p>The identifier of the function's <a href=\"https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html\">runtime</a>.</p>"
      }
    },
```
That's why: before your PR, the object was only used by `FunctionConfiguration`.
Now it's used by `FunctionConfiguration` and `ListLayerVersionsRequest` and the second definition is used.

File: src/Service/Lambda/tests/Integration/LambdaClientTest.php [hunk 2]
```diff
@@ -38,6 +41,20 @@ public function testAddLayerVersionPermission(): void
         self::assertStringContainsString('change it', $result->getRevisionId());
     }
 
+    public function testDeleteFunction(): void
+    {
+        self::markTestSkipped('The Lambda Docker image does not implement DeleteFunction.');
```
Comment (jderusse) [inline src/Service/Lambda/tests/Integration/LambdaClientTest.php:46]: ho. Really? :(

**Review by jderusse — APPROVED** [review]

File: src/Service/Lambda/tests/Integration/LambdaClientTest.php (hunk 2 above)
Comment (Nyholm) [reply src/Service/Lambda/tests/Integration/LambdaClientTest.php:46]: I got a 405

**Pull request merged**